File: Sorting_Algo/insertionsort.py
#Insertion sort-O(n^2)

a=[]
for i in range(int(input())):
	x=int(input())
	a.append(x)


for i in range(1,len(a)):
	key=a[i]
	j=i-1
	while(j>=0 and key<a[j]):
		a[j+1]=a[j]
		j=j-1
	a[j+1]=key
print("After Sorting")
print(a)

File: Sorting_Algo/Insertionsort.java
import java.util.Scanner;
import java.util.Arrays;
public class Insertionsort {
    public void insertionSort(int arr[]){
        int n=arr.length;
        for(int i=1;i<n;i++){
            int key=arr[i];
            int j=i-1;
            while(j>=0 && arr[j]>key){
                arr[j+1]=arr[j];
                j=j-1;
            }
            arr[j+1]=key;
        }
    }

    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int n;
        int a[];
        n=sc.nextInt();
        a=new int[n];
        for(int i=0;i<n;i++){
            a[i]=sc.nextInt();
        }
        sc.close();
        Insertionsort ob=new Insertionsort();
        ob.insertionSort(a);
        //printing after sorting
        System.out.println(Arrays.toString(a));
    }
}
